src/jit/compiler.cpp

```diff
@@ -11085,3 +11085,231 @@ bool Compiler::killGCRefs(GenTree* tree)
     }
     return false;
 }
+
+//------------------------------------------------------------------------
+// getSiVarLoc: Returns a "siVarLoc" instance representing the place where the variable 
+// is given its description, "baseReg", and "offset" (if needed).
+//
+// Arguments:
+//    varDsc    - a "LclVarDsc *" to the variable it is desired to build the "siVarLoc".
+//    baseReg   - a "regNumber" use as a base for the offset.
+//    offset    - a signed amount of bytes distance from "baseReg" for the position of the variable.
+//
+// Return Value:
+//    A "siVarLoc" filled with the correct case struct fields for the variable, which could live
+//    in a register, an stack position, or a combination of both.
+//
+// Assumptions:
+//    No new memory is allocated on this method.
+//    No changes on "varDsc" are applied.
+//
+// Notes:
+//    The "baseReg" and "offset" cannot be obtained from "varDsc".
+//    It calls "fillRegisterVarLoc" or "fillStackVarLoc" depending on if the "varDsc" argument.
+//    "lvIsInReg", so their notes and assumptions are held.
+Compiler::siVarLoc Compiler::getSiVarLoc(LclVarDsc* varDsc, regNumber baseReg, signed offset) const
```

Comment:
Why is this returning `siVarLoc` by value but the fill functions use an out parameter?

Reply:
I don't have a final conclusion. Do we have a preference?
I find copy by value easier to read, but passing it as a reference is more performant.

Reply:
Hmm, I'm not familiar with this code but from a quick look I think I might chose to do neither. Moving all this initialization code to a `siVarLoc` constructor might make more sense.

Otherwise I think I'd just return the struct. It's more or less the same thing performance wise because returning a (large enough struct) is done via a hidden out argument. In the worst case you get an extra copy but this struct is small enough that it doesn't really matter.

Reply:
A `siVarLoc::siVarLoc(const LclVarDsc* var, const siScope* scope)` constructor seems quite reasonable for all this code.

And `siVarLoc(const psiScope* scope)` for the other use of `siVarLoc`.

Reply:
I like this idea. The only problem is that siScope is protected inside CodeGen class and siVarLoc is used in other structs of compiler.h

Reply:
Well, the whole thing looks like a mess. As far as I can tell, it's all codegen related. It's just that for some reason `eeSetLVcount`, `eeSetLVinfo`, `eeSetLVdone` are in `Compiler` when they could have been in `CodeGen`. I won't even comment on stuff like `(ICorDebugInfo::NativeVarInfo*)eeVars` in `eeSetLVdone`.

Probably the simplest thing to do is to move `siVarLoc` from `Compiler` to `CodeGenInterface`.

Reply:
I have moved siVarLoc and siVarLocTypes to CodeGenInterface and the implementations of siVarLoc method/constructor to scopeinfo.cpp